documents/ARCHITECTURE.md

```diff
@@ -0,0 +1,80 @@
+# Stickian Empire - The Architecture
+
+This document holds the key architectural components that the developers of the Stickian Empire game follow. The reading is dense, and the writer is denser, but don't worry, nothing goes over his head, his reflexes are too quick.
+
+Take with a hint of salt any point given as a draft with the status :page_facing_up:.
+
+Work in progress... :warning: :construction_worker:
+
+## Index
+
+1. [High-Level Architecture](#high-level-architecture)
+2. [Tools](#tools)
+3. [Back-end Architecture](#back-end-architecture)
+4. [The NoSQL Approach](#the-nosql-approach)
+5. [Real(bothersome)-time problem](#realbothersome-time-problem)
+6. ["Micro" Front-ends](#micro-front-ends)
+7. [Front-end state management](#front-end-state-management)
+
+## High-Level Architecture
+
+To be able to provide fun to thousands, if millions, of (long distant future) players, a robust and future proof architecture is necessary. This first High-Level approach is to guide the developers and contributors to our final goal.
+
+```mermaid
+flowchart LR
+    subgraph frontend
+        home[Login/Home screen]
+        game[Game screen]
+        game[Game screen]
+        game[Game screen]
+        chat[Chat view]
+    end
+    subgraph backend
+        home --- auths
+        auths[Authentication service] --- relationaldb
+        game --- gameb1
+        game --- gameb2
+        game --- gameb3
+        gameb1[Game service X] --- gamedb
+        gameb2[Game service Y] --- gamedb
+        gameb3[Game service Z] --- gamedb
+        chat --- chatb
+        chatb[Chat service] --- relationaldb
+        gamedb[(NoSQL Database)]
+        relationaldb[(Relational Database)]
+    end
+```
```

Comment:
Do not forget that, later on, we may require a forum (with tutorials, player discussions, patch announcements), tickets' system (for reports and bugs) and maybe other features. These should stay at a sequential database I guess.

Reply:
Agreed that the relational database should be used for these features. Initially the announcements could be placed dynamically on this "Home screen", but later we can take other approaches.